seqr/views/apis/summary_data_api.py

```diff
@@ -113,29 +113,34 @@ def bulk_update_family_analysed_by(request):
         return create_json_response({'error': 'Project and Family columns are required'}, status=400)
     families_data = [dict(zip(header, row)) for row in family_upload_data[1:]]
 
-    family_qs = [Q(family_id=row['family'], project__name=row['project']) for row in families_data]
-    family_filter_q = family_qs[0]
-    for f_q in family_qs[1:]:
-        family_filter_q |= f_q
-    families = Family.objects.filter(family_filter_q)
+    projects = set()
+    families = set()
+    requested_families = set()
+    for row in families_data:
+        projects.add(row['project'])
+        families.add(row['family'])
+        requested_families.add((row['project'], row['family']))
+
+    family_db_id_lookup = {
+        (f['project__name'], f['family_id']): f['id'] for f in Family.objects.filter(
+            family_id__in=families, project__name__in=projects).values('id', 'family_id', 'project__name')
```

Comment:
How about using `Concat` to match the `project name` and `family id` pairs?

Reply:
I'd rather explicitly use tuples though, what is the reason why formatted string would be better?

Reply:
I mean using the Django `Concat` to connect two fields and filter out extra families. It makes the following code cleaner.

Reply:
ohh you mean in the filter input not the values output. Yeah I like that!